md5sum_cli.cpp

```diff
@@ -0,0 +1,264 @@
+// Truncated version of the original `md5sum` program.
+//
+// Copyright (C) 2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010, 2011, 2012, 2013, 2014, 2015, 2016, 2017, 2018, 2019 Gregory W. Chicares.
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 2 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; ifnot, write to the Free Software Foundation,
+// Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+//
+// http://savannah.nongnu.org/projects/lmi
+// email: <gchicares@sbcglobal.net>
+// snail: Chicares, 186 Belle Woods Drive, Glastonbury CT 06033, USA
+
+#include "pchfile.hpp"
+
+#include "md5sum.hpp"
+#include "getopt.hpp"
+#include "main_common.hpp"
+
+#include <iostream>
+
+int usage(int status)
+{
+    if(status != EXIT_SUCCESS)
+        {
+        std::cerr << "Try 'md5sum --help' for more information." << std::endl;
+        }
+    else
+        {
+        std::cout
+            << "Usage: md5sum [OPTION]... [FILE]..." << std::endl
```

Comment:
Use of `std::endl` here looks unnecessary, we don't need to flash the stream after every line, so what's wrong with just `\n`?

I'd also use C++11 raw literal to just output the entire message at once instead of using plenty of `<<` operators.